app/src/main/java/Fragments/SearchFragment.java

```diff
@@ -26,12 +65,169 @@ public static SearchFragment newInstance() {
         return fragment;
     }
 
+    //Still need to comment code! Someone remind me if I forget when I submit a pull request!
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_search, container, false);
+
+        searchNameET = (EditText) view.findViewById(R.id.search_name);
+        minPriceET = (EditText) view.findViewById(R.id.search_minprice);
+        maxPriceET = (EditText) view.findViewById(R.id.search_maxprice);
+
+        categorySpinner = (Spinner) view.findViewById(R.id.search_category);
+        subCategorySpinner = (Spinner) view.findViewById(R.id.search_subCategory);
+
+        searchButton = (Button) view.findViewById(R.id.search_submit);
+
+
+        categoriesRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                categoryOptions = getArrayFromSnapshot(dataSnapshot);
+
+                ArrayAdapter adapter = new ArrayAdapter(getContext(), android.R.layout.simple_spinner_item, categoryOptions); //this, android.R.layout.simple_spinner_item, categoryOptions);
+
+                categorySpinner.setAdapter(adapter);
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+
+        justCreatedFlagC = true;
+        selectedCategory = -1;
+        categorySpinner.setOnItemSelectedListener(
+                new AdapterView.OnItemSelectedListener() {
+                    @Override
+                    public void onItemSelected(AdapterView<?> adapterView, View view, int category, long id) {
+                        if(justCreatedFlagC){
+                            justCreatedFlagC = false;
```

Comment:
I have my guesses for what this is for but can you confirm the purpose of justCreatedFlagC. Is it because the first item is selected by default and you want to have it ignored?

Reply:
I added the flag because onItemSelected is run on start. I don't actually need (or want) it to do anything at this point, so I implemented the flag to make sure it doesn't.